src/components/Integrations/BigQuery/ControlPanel/ControlPanel.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import { useIntl } from 'react-intl';
+import MultiInputContainer from './MultiInputContainer';
+import { FormattedMessageMarkdown } from '../../../../i18n/FormattedMessageMarkdown';
+import { VerticalLine } from './style/styles.js';
+import datastudiogif from '../../../../img/google-data-studio.gif';
+
+interface IControlPanelProps {
+  values?: string[];
+  onChange: (newValues: string[]) => void;
+}
+
+const ControlPanel: React.FunctionComponent<IControlPanelProps> = (props) => {
+  const intl = useIntl();
+  const i18 = useCallback(
+    (id: string, values?: any) => {
+      return intl.formatMessage({ id: id }, values);
+    },
+    [intl],
+  );
+
+  const [values, setValues] = useState<string[]>([]);
+  const [message, setMessage] = useState(false);
+  const [saved, setSaved] = useState(false);
+
+  const onClickHandler = useCallback(
+    (isSaved: boolean) => {
+      if (isSaved) {
+        setMessage(isSaved);
+        setSaved(isSaved);
+      } else {
+        setMessage(isSaved);
+        setSaved(isSaved);
+      }
+    },
+    [setMessage, setSaved],
+  );
```

Comment:
What do you think about having two differnet functions, one for "save" and another one "cancel"?

Reply:
I do not understand. If `isSaved` is `true` run this:
```
setMessage(isSaved);
setSaved(isSaved);
```
but if it is false it also executes the same. It would be better to remove the if-else block and just write:
```
(isSaved: boolean) => {
    setMessage(isSaved);
    setSaved(isSaved);
}
```

I think `setSaved` and `setMessage` are not necessary to add them to the **dependency array**.

Reply:
Suggest accepted and applied !!!

Reply:
> What do you think about having two differnet functions, one for "save" and another one "cancel"?

yes i agree with you !!!